base/src/main/java/in/droidcon/base/repo/remote/ServiceFactory.kt

```diff
@@ -0,0 +1,68 @@
+package `in`.droidcon.base.repo.remote
+
+import com.google.gson.FieldNamingPolicy
+import com.google.gson.Gson
+import com.google.gson.GsonBuilder
+import okhttp3.OkHttpClient
+import okhttp3.logging.HttpLoggingInterceptor
+import retrofit2.Retrofit
+import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
+import retrofit2.converter.gson.GsonConverterFactory
+import java.util.concurrent.TimeUnit
+
+object ServiceFactory {
+    /**
+     * Provide "make" methods to create instances of [ServiceClass]
+     * and its related dependencies, such as OkHttpClient, Gson, etc.
+     */
+    inline fun <reified ServiceClass> makeService(isDebug: Boolean): ServiceClass = makeService(
+        isDebug,
+        makeOkHttpClient(
+            makeLoggingInterceptor(isDebug)
+        ), makeGson()
+    )
+
+    inline fun <reified ServiceClass> makeService(
+        isDebug: Boolean,
+        okHttpClient: OkHttpClient,
+        gson: Gson
+    ): ServiceClass = makeRetrofit(isDebug, okHttpClient, gson).create(ServiceClass::class.java)
+
+    fun makeRetrofit(
+        isDebug: Boolean,
+        okHttpClient: OkHttpClient,
+        gson: Gson
+    ): Retrofit = Retrofit.Builder()
+        .baseUrl(
+            if (isDebug) {
```

Comment:
Would be better if the baseUrl is configured as a part of the gradle configuration so that different product flavours could have different configurations. Since we are doing this to showcase various best practices, feel it would be good to have that kind of sample in here.

Reply:
Done